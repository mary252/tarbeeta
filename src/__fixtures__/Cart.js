export const CartList = [
  {
    avatar: "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
    cart_id: 107,
    name: "Lacoste",
    products: [
      {
        abbreviation: "XL",
        colour_id: "1",
        colour_img: "https://static.tarbeeta.com/images/colors/1/white.svg",
        currency: "ج.م",
        img:
          "https://static.tarbeeta.com/images/products/25/colour/1/1557599974364.jpg",
        price: "1250",
        product_id: "25",
        qty: 1,
        size_name: "إكس لارچ",
        stock: "5",
        title: "Sleeve crew t-shirt",
        variation_id: "153"
      },
      {
        abbreviation: "3XL",
        colour_id: "1",
        colour_img: "https://static.tarbeeta.com/images/colors/1/white.svg",
        currency: "ج.م",
        img:
          "https://static.tarbeeta.com/images/products/23/colour/1/1557517342506.jpg",
        price: "3250",
        product_id: "23",
        qty: 1,
        size_name: "Triple X-Large",
        stock: "3",
        title: "Lacoste Ortholite Rene",
        variation_id: "88"
      }
    ],
    shop_id: 57
  },
  {
    avatar: "https://static.tarbeeta.com/images/shops/58/1557676045436.jpeg",
    cart_id: 107,
    name: "IBIS",
    products: [
      {
        abbreviation: "OS",
        colour_id: "1",
        colour_img: "https://static.tarbeeta.com/images/colors/1/white.svg",
        currency: "ج.م",
        img:
          "https://static.tarbeeta.com/images/products/46/colour/1/1557928705953.jpg",
        price: "289",
        product_id: "46",
        qty: 1,
        size_name: "مقاس  واحد",
        stock: "1",
        title: "Infinite Train",
        variation_id: "294"
      }
    ],
    shop_id: 58
  },
  {
    avatar: "https://static.tarbeeta.com/images/shops/59/1557848536743.png",
    cart_id: 107,
    name: "Magma Sports Wear",
    products: [
      {
        abbreviation: "OS",
        colour_id: "1",
        colour_img: "https://static.tarbeeta.com/images/colors/1/white.svg",
        currency: "ج.م",
        img:
          "https://static.tarbeeta.com/images/products/46/colour/1/1557928705953.jpg",
        price: "289",
        product_id: "46",
        qty: 1,
        size_name: "مقاس  واحد",
        stock: "1",
        title: "Infinite Train",
        variation_id: "294"
      }
    ],
    shop_id: 59
  }
];

export const total = {
  currency: "ج.م",
  currency_id: 1,
  total: 8584
};
