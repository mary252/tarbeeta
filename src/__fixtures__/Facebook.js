export const faceRes = {
  data: {
    avatar: null,
    mobile_verified: 1,
    shop_id: "10",
    shop_username: "testShop",
    user_id: 152,
    verified: 1
  },
  token:
    "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTUyLCJyb2xlX2lkIjozLCJpYXQiOjE1NjMzNjA3MzQsImV4cCI6MTU3MTEzNjczNH0.CcoYVKrR2jdRCUloYoB53iBtNR78xoR-nzARch-hDuBqQSs5HHPIs4S5Gg7ejAlB7GJCTqPxSHTH4V102rDEwBmQQDEc9FOvkQg0Rz1kCDYC9ev3Q5uylwFQgmovXnBARGCs-tfhgJmOMsOhfIemQr5VKRpdg3w96hGXnwXGpxA"
};
