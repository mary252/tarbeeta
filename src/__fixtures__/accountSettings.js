export const data = {
    firstname: "Mohamed",
    lastname: "Shawky",
    mobile: "01142290603",
    email: "mohamedshawkybayoumi@gmail.com"
}