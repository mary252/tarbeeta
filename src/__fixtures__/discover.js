export const data = [
    {
        avatar: "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
        colour_id: 11,
        currency: "ج.م",
        description: "Lacoste sleeve crew t-shirt for Men",
        id: 25,
        image: "https://static.tarbeeta.com/images/products/25/colour/11/1557599974322.jpg",
        is_liked: 0,
        is_mine: 0,
        price: 1250,
        shop_id: 57,
        shop_name: "Lacoste",
        shop_username: "lacoste",
        stock: 38,
        title: "Sleeve crew t-shirt",
        variation_id: 1
    }
];