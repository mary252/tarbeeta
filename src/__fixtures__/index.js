export * from "./FavoritesList";
export * from "./Cart";
export * from "./discover";
export * from "./Facebook";
