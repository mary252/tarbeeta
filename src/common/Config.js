// export const baseURL = "https://api.tarbeeta.com/v1/";
// export const v2_url = "https://api.tarbeeta.com/v2/";
export const v2_url = "https://staging.tarbeeta.com/v2/";
export const baseURL = "https://staging.api.tarbeeta.com/v1/";
export const socketUrl = "https://socket.tarbeeta.com:8878";
// export const baseURL = "http://192.168.1.15:3000/v1/";
// export const baseURL = "http://197.167.99.223:3126/v1/";
 //export const baseURL = "http://msi-laptop:3000/v1/";
export const assetUrl = "https://static.tarbeeta.com/images/";
export const baseUrlStatic = "https://www.tarbeeta.com";

//export const google_capcha="6Lf34qkUAAAAAKBh0Qs_9OZrBeZhikwqF6L7G62p"
// export const google_capcha="6LfE5akUAAAAAHC3KCKqJh1NGup58xJwQbjZxYgg"
export const google_capcha = "6LfXz6oUAAAAABbSRcTAPsK971UiPQZ6CpSSPWr4";
export const EN = 1;
export const AR = 2;
export const EG = 1;
export const DEFAULT_LOCALE = "ar";

// Facebook
export const FACEBOOK_APP_ID = "520130485093535";
export const FACEBOOK_PAGE_ID = "1996928467010956";
export const FACEBOOK_CHAT_POPUP_THEME_COLOR = "#5a9ae5";

export const PRODUCT_SLIDER_LIMIT = 18;

export const ORDER_STATUS_FAILED = 1;
export const ORDER_STATUS_PLACED = 2;
export const ORDER_STATUS_PAID = 3;
export const ORDER_STATUS_PENDING_PAYMENT = 4;
export const ORDER_STATUS_PENDING_SHIPPING = 5;
export const ORDER_STATUS_ON_THE_WAY = 6;
export const ORDER_STATUS_DELIVERED = 7;
export const ORDER_STATUS_CANCELLED = 8;
export const ORDER_STATUS_PENDING = 9;

export const APPROVED_PRODUCT = 4;

export const CASH_ON_DELEVERY = 4;
export const PAY_WALLET = 3;
export const MASTER_CARD = 1;
export const KIOSK = 2;

export const SORT_MOST_RECENT = 1;
export const SORT_OLD_TO_NEW = 2;
export const SORT_HIGHEST_PRICE = 3;
export const SORT_LOWEST_PRICE = 4;
export const SORT_RATING = 5;
export const CARD_PAYMENT_IFRAME = `https://accept.paymobsolutions.com/api/acceptance/iframes/`;
export const EN_FRAME_ID = 7340;
export const AR_FRAME_ID = 7459;
export const pending_product = 1;
