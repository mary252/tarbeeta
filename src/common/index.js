export * from "./Images";
export * from "./Config";
