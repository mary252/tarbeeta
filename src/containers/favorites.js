import React, { Component } from "react";
import { Page, BreadCrumb } from "../components";
import ProductsGrid from "./Products/Grid";
import * as actions from "../actions";
import { connect } from "react-redux";
import MessengerCustomerChat from "react-messenger-customer-chat";

import { getLikes } from "../services";
import {
  FACEBOOK_CHAT_POPUP_THEME_COLOR,
  FACEBOOK_APP_ID,
  FACEBOOK_PAGE_ID
} from "../common";

export class Favorites extends Component {
  state = {
    products: [],
    loading: false,
    pageTitle: "",
    showMore: false,
    page: 1,
    count: 0
  };

  componentDidMount = async () => {
    this.setState({ loading: true });

    await this.getLikesProducts();

    this.setState({ loading: false });
  };

  getLikesProducts = async willMerge => {
    try {
      this.setState({ is_busy: true });
      let page = this.state.page,
        lang_id = this.props.lang_id;

      let res = await getLikes(lang_id, page);

      this.setState(({ products }) => ({
        products: willMerge ? [...products, ...res.data] : res.data,
        showMore: res.showMore,
        is_busy: false,
        count: res.count
      }));
    } catch (e) {}
  };

  likePro = variation_id => {
    this.setState(state => ({
      products: state.products.filter(el => el.variation_id !== variation_id)
    }));
  };

  openProductDetails = (id, colId) => {
    this.props.TogglePopup(id, this.props.lang_id, colId);
  };

  loadMore = () =>
    this.setState(
      state => ({ page: state.page + 1 }),
      () => this.getLikesProducts(true)
    );

  render() {
    const {
      state: {
        products = [],
        filtersBar,
        pageTitle,
        is_busy,
        loading,
        showMore,
        count
      },
      props: { lang, location, locale },
      loadMore
    } = this;

    return (
      <Page
        title={locale.Favorite_title}
        description={locale.Facvorites_desc}
        {...this.props}
      >
        <div className="section">
          <div className="container">
            <div>
              <MessengerCustomerChat
                pageId={FACEBOOK_PAGE_ID}
                greeting_dialog_display="hide"
                appId={FACEBOOK_APP_ID}
                themeColor={FACEBOOK_CHAT_POPUP_THEME_COLOR}
                language={this.props.lang == "en" ? "en_US" : "ar_AR"}
                greetingDialogDelay={0}
                minimized={true}
                shouldShowDialog={false}
                loggedInGreeting={
                  this.props.locale.facebook_chat_greeting_message
                }
                loggedOutGreeting={
                  this.props.locale.facebook_chat_greeting_message
                }
              />
            </div>
            <BreadCrumb
              history={[
                {
                  name: locale.tarbeeta,
                  href: `/${lang}`
                }
              ]}
              activeRoute={locale.Favorite_title}
            />

            <ProductsGrid
              count={count}
              lang={this.props.lang}
              locale={this.props.locale}
              loading={loading}
              is_busy={is_busy}
              loadMore={loadMore}
              showMore={showMore}
              openProductDetails={this.openProductDetails}
              nosorting
              sort={this.sort}
              title={pageTitle}
              withseller={true}
              products={products}
              goToDetails={this.goToDetails}
              filtersBar={filtersBar}
              select_filter={this.select_filter}
              match={this.props.match}
              onLiked={this.likePro}
              lang={this.props.lang}
              Favourites
              small_cards
            />
          </div>
        </div>
      </Page>
    );
  }
}

export default connect(
  null,
  actions
)(Favorites);
