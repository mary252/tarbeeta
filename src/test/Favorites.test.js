import React from "react";
import { shallow } from "enzyme";
import {Favorites} from "../containers/favorites";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { baseURL } from "../common/Config";
import axios from "axios"; // v0.15.3
import httpAdapter from "axios/lib/adapters/http";
import nock from "nock";
import { FavoritesList } from "../__fixtures__";

configure({ adapter: new Adapter() });

const match = {
  params: {
    lang: "en" //any id you want to set
  }
};
const location = {
  search: ""
};

const host = baseURL;

axios.defaults.host = host;
axios.defaults.adapter = httpAdapter;
let query = {
  lang_id: 1
};

const favorites = shallow(<Favorites match={match} location={location} />);
let instance = favorites.instance();

describe("Favorites Container", () => {

  it("Should Fetch Faqs And Update the State", async() => {
   nock(host)
  .get("/product/liked")
  .query(query)
  .reply(200, {
    data: FavoritesList
  });
    await instance.componentDidMount()
    expect(instance.state.products).toEqual(FavoritesList);

    
  });

  it("Should On Like Remove Product", () => {
    instance.setState(() => ({ products: FavoritesList }));
    instance.likePro(234);

    expect(instance.state.products.length).toEqual(2);
  });

  it("  Faqs container exists", () => {
    expect(favorites.exists()).toEqual(true);
  });

  it("renders correctly", () => {
    expect(favorites).toMatchSnapshot();
  });
});
