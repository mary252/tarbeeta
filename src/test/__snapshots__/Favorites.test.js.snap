// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Container renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    location={
      Object {
        "search": "",
      }
    }
    match={
      Object {
        "params": Object {
          "lang": "en",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="section"
      >
        <div
          className="container"
        >
          <BreadCrumb
            activeRoute="Fashion"
            history={
              Array [
                Object {
                  "href": "/en",
                  "name": "Tarbeeta",
                },
              ]
            }
          />
          <Connect(ProductsGrid)
            is_busy={true}
            loadMore={[Function]}
            match={
              Object {
                "params": Object {
                  "lang": "en",
                },
              }
            }
            nosorting={true}
            onLiked={[Function]}
            openProductDetails={[Function]}
            products={
              Array [
                Object {
                  "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                  "colour_id": 3,
                  "currency": "EGP",
                  "description": "Quality polo t-shirts for Men",
                  "id": 22,
                  "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                  "is_liked": 1,
                  "is_mine": 0,
                  "price": 1125,
                  "shop_id": 57,
                  "shop_name": "Lacoste",
                  "shop_username": "lacoste",
                  "stock": 32,
                  "title": "Polo t-shirt for Men",
                  "variation_id": 74,
                },
                Object {
                  "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                  "colour_id": 9,
                  "currency": "EGP",
                  "description": "Quality polo t-shirts for Men",
                  "id": 22,
                  "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                  "is_liked": 1,
                  "is_mine": 0,
                  "price": 1125,
                  "shop_id": 57,
                  "shop_name": "Lacoste",
                  "shop_username": "lacoste",
                  "stock": 33,
                  "title": "Polo t-shirt for Men",
                  "variation_id": 55,
                },
              ]
            }
            showMore={false}
            title=""
            withseller={true}
          />
        </div>
      </div>,
      "location": Object {
        "search": "",
      },
      "match": Object {
        "params": Object {
          "lang": "en",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <BreadCrumb
            activeRoute="Fashion"
            history={
              Array [
                Object {
                  "href": "/en",
                  "name": "Tarbeeta",
                },
              ]
            }
          />
          <Connect(ProductsGrid)
            is_busy={true}
            loadMore={[Function]}
            match={
              Object {
                "params": Object {
                  "lang": "en",
                },
              }
            }
            nosorting={true}
            onLiked={[Function]}
            openProductDetails={[Function]}
            products={
              Array [
                Object {
                  "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                  "colour_id": 3,
                  "currency": "EGP",
                  "description": "Quality polo t-shirts for Men",
                  "id": 22,
                  "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                  "is_liked": 1,
                  "is_mine": 0,
                  "price": 1125,
                  "shop_id": 57,
                  "shop_name": "Lacoste",
                  "shop_username": "lacoste",
                  "stock": 32,
                  "title": "Polo t-shirt for Men",
                  "variation_id": 74,
                },
                Object {
                  "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                  "colour_id": 9,
                  "currency": "EGP",
                  "description": "Quality polo t-shirts for Men",
                  "id": 22,
                  "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                  "is_liked": 1,
                  "is_mine": 0,
                  "price": 1125,
                  "shop_id": 57,
                  "shop_name": "Lacoste",
                  "shop_username": "lacoste",
                  "stock": 33,
                  "title": "Polo t-shirt for Men",
                  "variation_id": 55,
                },
              ]
            }
            showMore={false}
            title=""
            withseller={true}
          />
        </div>,
        "className": "section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BreadCrumb
              activeRoute="Fashion"
              history={
                Array [
                  Object {
                    "href": "/en",
                    "name": "Tarbeeta",
                  },
                ]
              }
            />,
            <Connect(ProductsGrid)
              is_busy={true}
              loadMore={[Function]}
              match={
                Object {
                  "params": Object {
                    "lang": "en",
                  },
                }
              }
              nosorting={true}
              onLiked={[Function]}
              openProductDetails={[Function]}
              products={
                Array [
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 3,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 32,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 74,
                  },
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 9,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 33,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 55,
                  },
                ]
              }
              showMore={false}
              title=""
              withseller={true}
            />,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeRoute": "Fashion",
              "history": Array [
                Object {
                  "href": "/en",
                  "name": "Tarbeeta",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "filtersBar": undefined,
              "goToDetails": undefined,
              "is_busy": true,
              "loadMore": [Function],
              "match": Object {
                "params": Object {
                  "lang": "en",
                },
              },
              "nosorting": true,
              "onLiked": [Function],
              "openProductDetails": [Function],
              "products": Array [
                Object {
                  "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                  "colour_id": 3,
                  "currency": "EGP",
                  "description": "Quality polo t-shirts for Men",
                  "id": 22,
                  "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                  "is_liked": 1,
                  "is_mine": 0,
                  "price": 1125,
                  "shop_id": 57,
                  "shop_name": "Lacoste",
                  "shop_username": "lacoste",
                  "stock": 32,
                  "title": "Polo t-shirt for Men",
                  "variation_id": 74,
                },
                Object {
                  "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                  "colour_id": 9,
                  "currency": "EGP",
                  "description": "Quality polo t-shirts for Men",
                  "id": 22,
                  "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                  "is_liked": 1,
                  "is_mine": 0,
                  "price": 1125,
                  "shop_id": 57,
                  "shop_name": "Lacoste",
                  "shop_username": "lacoste",
                  "stock": 33,
                  "title": "Polo t-shirt for Men",
                  "variation_id": 55,
                },
              ],
              "select_filter": undefined,
              "showMore": false,
              "sort": undefined,
              "title": "",
              "withseller": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="section"
        >
          <div
            className="container"
          >
            <BreadCrumb
              activeRoute="Fashion"
              history={
                Array [
                  Object {
                    "href": "/en",
                    "name": "Tarbeeta",
                  },
                ]
              }
            />
            <Connect(ProductsGrid)
              is_busy={true}
              loadMore={[Function]}
              match={
                Object {
                  "params": Object {
                    "lang": "en",
                  },
                }
              }
              nosorting={true}
              onLiked={[Function]}
              openProductDetails={[Function]}
              products={
                Array [
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 3,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 32,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 74,
                  },
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 9,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 33,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 55,
                  },
                ]
              }
              showMore={false}
              title=""
              withseller={true}
            />
          </div>
        </div>,
        "location": Object {
          "search": "",
        },
        "match": Object {
          "params": Object {
            "lang": "en",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <BreadCrumb
              activeRoute="Fashion"
              history={
                Array [
                  Object {
                    "href": "/en",
                    "name": "Tarbeeta",
                  },
                ]
              }
            />
            <Connect(ProductsGrid)
              is_busy={true}
              loadMore={[Function]}
              match={
                Object {
                  "params": Object {
                    "lang": "en",
                  },
                }
              }
              nosorting={true}
              onLiked={[Function]}
              openProductDetails={[Function]}
              products={
                Array [
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 3,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 32,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 74,
                  },
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 9,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 33,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 55,
                  },
                ]
              }
              showMore={false}
              title=""
              withseller={true}
            />
          </div>,
          "className": "section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BreadCrumb
                activeRoute="Fashion"
                history={
                  Array [
                    Object {
                      "href": "/en",
                      "name": "Tarbeeta",
                    },
                  ]
                }
              />,
              <Connect(ProductsGrid)
                is_busy={true}
                loadMore={[Function]}
                match={
                  Object {
                    "params": Object {
                      "lang": "en",
                    },
                  }
                }
                nosorting={true}
                onLiked={[Function]}
                openProductDetails={[Function]}
                products={
                  Array [
                    Object {
                      "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                      "colour_id": 3,
                      "currency": "EGP",
                      "description": "Quality polo t-shirts for Men",
                      "id": 22,
                      "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                      "is_liked": 1,
                      "is_mine": 0,
                      "price": 1125,
                      "shop_id": 57,
                      "shop_name": "Lacoste",
                      "shop_username": "lacoste",
                      "stock": 32,
                      "title": "Polo t-shirt for Men",
                      "variation_id": 74,
                    },
                    Object {
                      "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                      "colour_id": 9,
                      "currency": "EGP",
                      "description": "Quality polo t-shirts for Men",
                      "id": 22,
                      "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                      "is_liked": 1,
                      "is_mine": 0,
                      "price": 1125,
                      "shop_id": 57,
                      "shop_name": "Lacoste",
                      "shop_username": "lacoste",
                      "stock": 33,
                      "title": "Polo t-shirt for Men",
                      "variation_id": 55,
                    },
                  ]
                }
                showMore={false}
                title=""
                withseller={true}
              />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeRoute": "Fashion",
                "history": Array [
                  Object {
                    "href": "/en",
                    "name": "Tarbeeta",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "filtersBar": undefined,
                "goToDetails": undefined,
                "is_busy": true,
                "loadMore": [Function],
                "match": Object {
                  "params": Object {
                    "lang": "en",
                  },
                },
                "nosorting": true,
                "onLiked": [Function],
                "openProductDetails": [Function],
                "products": Array [
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 3,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/3/1557515144829.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 32,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 74,
                  },
                  Object {
                    "avatar": "https://static.tarbeeta.com/images/shops/57/1557514373795.jpg",
                    "colour_id": 9,
                    "currency": "EGP",
                    "description": "Quality polo t-shirts for Men",
                    "id": 22,
                    "image": "https://static.tarbeeta.com/images/products/22/colour/9/1557515144750.jpg",
                    "is_liked": 1,
                    "is_mine": 0,
                    "price": 1125,
                    "shop_id": 57,
                    "shop_name": "Lacoste",
                    "shop_username": "lacoste",
                    "stock": 33,
                    "title": "Polo t-shirt for Men",
                    "variation_id": 55,
                  },
                ],
                "select_filter": undefined,
                "showMore": false,
                "sort": undefined,
                "title": "",
                "withseller": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
